File: telegram-bot-bumblebee-core/src/main/java/com/github/bumblebee/command/currency/CurrencyUSDActualExchangeRateCommand.java
package com.github.bumblebee.command.currency;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

import com.github.bumblebee.command.currency.domain.SupportedCurrency;
import com.github.bumblebee.command.currency.service.BYRExchangeRateRetrieveService;
import com.github.bumblebee.command.currency.service.CurrencyBidEvalService;
import com.github.bumblebee.service.RandomPhraseService;

import com.github.telegram.api.BotApi;
import com.github.telegram.domain.Update;

@Component
public class CurrencyUSDActualExchangeRateCommand extends CurrencyActualExchangeRateCommand {

    @Autowired
    public CurrencyUSDActualExchangeRateCommand(BotApi botApi, RandomPhraseService randomPhrase, CurrencyBidEvalService bidEvalService, BYRExchangeRateRetrieveService exchangeRateRetrieveService) {
        super(botApi, randomPhrase, bidEvalService, exchangeRateRetrieveService);
    }

    @Override
    public void handleCommand(Update update, long chatId, String argument) {
        super.handleCommand(update, chatId, SupportedCurrency.USD.name());
    }
}

File: telegram-bot-bumblebee-core/src/main/kotlin/com/github/bumblebee/command/currency/CurrencyUSDActualExchangeRateCommand.kt
package com.github.bumblebee.command.currency

import com.github.bumblebee.command.currency.domain.SupportedCurrency
import com.github.bumblebee.command.currency.service.BYRExchangeRateRetrieveService
import com.github.bumblebee.service.RandomPhraseService
import com.github.telegram.api.BotApi
import com.github.telegram.domain.Update
import org.springframework.stereotype.Component

@Component
class CurrencyUSDActualExchangeRateCommand(
        botApi: BotApi,
        randomPhrase: RandomPhraseService,
        exchangeRateRetrieveService: BYRExchangeRateRetrieveService)
    : CurrencyActualExchangeRateCommand(botApi, randomPhrase, exchangeRateRetrieveService) {

    override fun handleCommand(update: Update, chatId: Long, argument: String?) {
        super.handleCommand(update, chatId, SupportedCurrency.USD.name)
    }
}
